docs/public/secure-registry.adoc

```diff
@@ -35,3 +35,30 @@ Please run `odo registry add <registry name> <registry URL> --token <token>` to
 - <registry name>: user-defined devfile registry name.
 - <registry URL>: the URL of GitHub repository that you create on step 1.
 - <token>: the personal access token that you create on step 2.
+
+== Steps for setting up a secure starter project when the secure starter project is hosted on a GitHub repository
+
+=== Step 1: Creating a new GitHub repository to host the secure starter project
+
+1. Please https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/creating-a-new-repository[create a new private or enterprise GitHub repository] and push the starter project to the created repository. The sample GitHub-hosted starter project can be found https://github.com/odo-devfiles/nodejs-ex[here].
+
+2. Ensure the `projects` section in the corresponding devfile of your secure devfile registry links to the secure starter project, for example:
+
+```
+projects:
+  - name: nodejs-starter
+    git:
+      location: "<secure starter project link>"
+```
+
+=== Step 2: Creating a personal access token to access the secure starter project
+Please https://docs.github.com/en/github/authenticating-to-github/creating-a-personal-access-token[create a personal access token], select `repo` as token scope.
+
+=== Step 3: Creating a devfile component from the secure devfile registry and downloading the secure starter project from the secure starter project host
+
```

Comment:
`Creating a devfile component from the secure devfile registry and downloading the secure starter project`